enterprise/app/api_keys/api_keys.tsx

```diff
@@ -576,3 +573,73 @@ function newFormState<T extends ApiKeyFields>(request: T): FormState<T> {
     request,
   };
 }
+
+interface ApiKeyFieldProps {
+  value: string;
+}
+
+interface ApiKeyFieldState {
+  value: string;
+  isCopied: boolean;
+  hideValue: boolean;
+  displayValue: string;
+}
+
+const ApiKeyFieldDefaultState: ApiKeyFieldState = {
+  value: "",
+  isCopied: false,
+  hideValue: true,
+  displayValue: "************",
+};
+
+class ApiKeyField extends React.Component<ApiKeyFieldProps, ApiKeyFieldState> {
+  state = ApiKeyFieldDefaultState;
+
+  componentDidMount() {
+    this.setState({ value: this.props.value });
+  }
+
+  // onClick handler function for the copy button
+  private handleCopyClick() {
+    const copyTextToClipboard = async (text: string) => {
+      if ("clipboard" in navigator) {
+        return await navigator.clipboard.writeText(text);
+      } else {
+        return document.execCommand("copy", true, text);
+      }
+    };
+
+    // Asynchronously call copyTextToClipboard
+    copyTextToClipboard(this.state.value).then(() => {
+      // If successful, update the isCopied state value
+      this.setState({ isCopied: true });
+      setTimeout(() => {
+        this.setState({ isCopied: false });
+      }, 1500);
```

Comment:
nit: rather than using state, you can also show a "success" banner, like `alert.success("Copied!")` - that would simplify the logic here since you don't need a timeout or dedicated UI state

Reply:
I still prefer having a ui state here to turn the button to a green check. 
It's a visual signal to tell people to stop clicking.
Alerting would require the same state to deduplicate multiple clicks within a small time window.

Would adding an alert plus the green check button be overkill? Let's find out! 🤔